CRM/Core/BAO/Dashboard.php

```diff
@@ -399,33 +399,20 @@ public static function addDashlet(&$params) {
     $dashlet = new CRM_Core_DAO_Dashboard();
 
     if (!$dashboardID) {
-
       // Assign domain before search to allow identical dashlets in different domains.
-      if (empty($params['domain_id'])) {
-        $dashlet->domain_id = CRM_Core_Config::domainID();
-      }
-      else {
-        $dashlet->domain_id = $params['domain_id'] ?? NULL;
-      }
+      $dashlet->domain_id = $params['domain_id'] ?? CRM_Core_Config::domainID();
 
       // Try and find an existing dashlet - it will be updated if found.
-      if (!empty($params['name'])) {
+      if (!empty($params['name']) || !empty($params['url'])) {
```

Comment:
@colemanw is this aiming to fix a bug or something, I'm not clear why we are condensing the if / else here, The code doesn't seem to be the same after

Reply:
It's essentially the same before/after. The only difference is that if both params are passed in they will both be taken into account in the search, which seems like an improvement.